src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -186,7 +186,20 @@ impl<'top> LazyRawBinaryValue_1_1<'top> {
 
     /// Helper method called by [`Self::read`]. Reads the current value as a bool.
     fn read_bool(&self) -> ValueParseResult<'top, BinaryEncoding_1_1> {
-        todo!();
+        debug_assert!(self.encoded_value.ion_type() == IonType::Bool);
+        let header = &self.encoded_value.header();
+        let representation = header.type_code();
+        let value = match (representation, header.length_code) {
+            (OpcodeType::Boolean, 0xE) => true,
+            (OpcodeType::Boolean, 0xF) => false,
+            invalid => {
+                return IonResult::decoding_error(format!(
+                    "found a boolean value with an illegal type code representation: {:?}",
+                    invalid
+                ))
```

Comment:
I assume that this branch can't be reached unless there's a bug, in which case I think it's ok for this to be an `unreachable!("...")` panic.